test/unit/pool/test-destroy-connection-timeout.js

```diff
@@ -0,0 +1,25 @@
+
+var assert     = require('assert');
+var common     = require('../../common');
+var Connection = common.Connection;
+var pool       = common.createPool({port: common.fakeServerPort});
+
+var server = common.createFakeServer();
+
+server.listen(common.fakeServerPort, function (err) {
+  assert.ifError(err);
+  pool.getConnection(function (err, connection) {
+    assert.ifError(err);
+    process.nextTick(function() {
+      connection.destroy();
+      pool.end(function (err) {
+        assert.ifError(err);
+        server.destroy();
+      })
+    })
+  })
+});
+
+setTimeout(function () {
+  throw new Error("PoolConnection.destroy left timers active after teardown")
+}, 5000).unref()
```

Comment:
> But that still feels like it's testing implementation details rather than user-visible behaviour.

That's no different than the test is now--it's basically testing some black magic where it is expecting the event loop to exit within 5 second (which can actually lead to false-positives on random machine/VM slowdowns during the test, as it's testing some weird action-at-a-distance).

Since the `enqueue` event is a public event and it doesn't suffer from the false positives, I think that would be an acceptable change.